mmv1/products/oracledatabase/AutonomousDatabase.yaml

```
@@ -0,0 +1,618 @@
+immutable: true
+timeouts:
+  insert_minutes: 240
+  update_minutes: 120
+  delete_minutes: 120
+import_format:
+  - 'projects/{{project}}/locations/{{location}}/autonomousDatabases/{{autonomous_database_id}}'
+autogen_async: true
+async:
+  actions: ['create', 'delete', 'update']
+  type: 'OpAsync'
+  operation:
+    base_url: '{{op_id}}'
+    path: 'name'
+    wait_ms: 1000
+    timeouts:
+      insert_minutes: 240
+      update_minutes: 120
+      delete_minutes: 120
+  result:
+    path: 'response'
+    resource_inside_response: true
+  error:
+    path: 'error'
+    message: 'message'
+custom_code:
+examples:
+  - name: 'oracledatabase_autonomous_database_basic'
+    primary_resource_id: 'myADB'
+    vars:
+      project: 'my-project'
+      autonomous_database_id: 'my-instance'
+    test_vars_overrides:
+      'project': '"oci-terraform-testing"'
+      'autonomous_database_id': '"my-adb-instance-id"'
+  - name: 'oracledatabase_autonomous_database_full'
+    primary_resource_id: 'myADB'
+    vars:
+      project: 'my-project'
+      autonomous_database_id: 'my-instance'
+    test_vars_overrides:
+      'project': '"oci-terraform-testing"'
+      'autonomous_database_id': '"my-adb-instance-id-2"'
+parameters:
+  - name: 'location'
+    type: String
+    description: 'Resource ID segment making up resource `name`. See documentation
+    for resource type `oracledatabase.googleapis.com/AutonomousDatabaseBackup`. '
+    url_param_only: true
+    required: true
+    immutable: true
+  - name: 'autonomousDatabaseId'
+    type: String
+    description: "The ID of the Autonomous Database to create. This value
+    is restricted\nto (^[a-z]([a-z0-9-]{0,61}[a-z0-9])?$) and must be a maximum of
+    63\ncharacters in length. The value must start with a letter and end with\na letter
+    or a number. "
+    url_param_only: true
+    required: true
+    immutable: true
+properties:
+  - name: 'name'
+    type: String
+    description: "Identifier. The name of the Autonomous Database resource in the
+     following format:\nprojects/{project}/locations/{region}/autonomousDatabases/{autonomous_database} "
+    output: true
+  - name: 'database'
+    type: String
+    description: "The name of the Autonomous Database. The database name must
+    be unique in\nthe project. The name must begin with a letter and can\ncontain
+    a maximum of 30 alphanumeric characters. "
+    required: true
+  - name: 'displayName'
+    type: String
+    description: "The display name for the Autonomous Database. The name does
+    not have to\nbe unique within your project. "
+    default_from_api: true
+  - name: 'entitlementId'
+    type: String
+    description: "The ID of the subscription entitlement associated with
+    the Autonomous\nDatabase. "
+    output: true
+  - name: 'adminPassword'
+    type: String
+    description: 'The password for the default ADMIN user. '
+    ignore_read: true
+  - name: 'properties'
+    type: NestedObject
```

```suggestion
    type: NestedObject
    required: true
```

Per https://github.com/GoogleCloudPlatform/magic-modules/pull/11960/files#r1798802833, this field is required since some of its fields are required at creation time.